config.lua

```diff
@@ -1,21 +1,45 @@
+---@class PostalConfig
+---@field versionCheck boolean Whether the server version check is enabled
+---@field text PostalTextConfig Configuration options for the text
+---@field blip PostalBlipConfig Configuration options for the blips
+---@field updateDelay number How often the current postal will be updated 
 config = {
-	versionCheck = true, -- enables version checking (if this is enabled and there is no new version it won't display a message anyways)
-	text = {
-		format = '~y~Nearest Postal~w~: %s (~g~%.2fm~w~)',
-		-- ScriptHook PLD Position
-		--posX = 0.225,
-		--posY = 0.963,
-		-- vMenu PLD Position
-		posX = 0.22,
-		posY = 0.963
-	},
-	blip = {
-		blipText = 'Postal Route %s',
-		sprite = 8,
-		color = 3, -- default 3 (light blue)
-		distToDelete = 100.0, -- in meters
-		deleteText = 'Route deleted',
-		drawRouteText = 'Drawing a route to %s',
-		notExistText = "That postal doesn't exist"
-	}
+    -- enables version checking (if this is enabled and there is no new version it won't display a message anyways)
+    versionCheck = true,
+
+    ---@class PostalTextConfig The nearest-postal text configuration
+    ---@field format string A Lua formatted string that will be shown to the player
+    ---@field posX number The on-screen X position
+    ---@field posY number The on-screen Y position
+    text = {
+        format = '~y~Nearest Postal~w~: %s (~g~%.2fm~w~)',
+        -- ScriptHook PLD Position
+        --posX = 0.225,
+        --posY = 0.963,
+        -- vMenu PLD Position
+        posX = 0.22,
+        posY = 0.963
+    },
+
+    ---@class PostalBlipConfig The nearest-postal blip configuration
+    ---@field blipText string A Lua formatted string that will be shown to the player
+    ---@field sprite number The sprite ID to display
+    ---@field color number The color ID to display
+    ---@field distToDelete number When a player is within this distance (meters), the blip is removed
+    ---@field deleteText string The text to show for clearing a route
+    ---@field drawRouteText string The text to show when drawing a route
+    ---@field notExistText string The text to show when a postal is not found
+    blip = {
+        blipText = 'Postal Route %s',
+        sprite = 8,
+        color = 3, -- default 3 (light blue)
+        distToDelete = 100.0, -- in meters
+        deleteText = 'Route deleted',
+        drawRouteText = 'Drawing a route to %s',
+        notExistText = "That postal doesn't exist"
+    },
+
+    --- The delay in milliseconds until the current postal code and text are updated
+    --- I wouldn't recommend anything lower than 50ms, as calculating the postal can be intensive.
+    updateDelay = 300,
```

Comment:
It's probably for the best to remove the classification comments from the configuration file as I'm sure people would probably get confused. First hand experience dealing with them lol

Reply:
I was pondering this as well, I think I will remove the luadoc in favor of individual comments for each. I will also handle fallback support for users with old `config.lua` files. Does `300ms` seem like an appropriate default to you though?

Reply:
`300ms` works for me 👍